compiler/src/dmd/dsymbolsem.d

```diff
@@ -7412,3 +7412,136 @@ private extern(C++) class SetFieldOffsetVisitor : Visitor
+    }
+
+override void visit(VisibilityDeclaration atbd)
+    {
+        if (atbd.pkg_identifiers)
+        {
+            dsymbolSemantic(atbd, sc);
+        }
+        sc = atbd.createNewScope(sc, sc.stc, sc.linkage, sc.cppmangle, atbd.visibility, 1, sc.aligndecl, sc.inlining);
+    }
+    /**
+     * Returns:
+     *   A copy of the parent scope, with `this` as `namespace` and C++ linkage
+     *///override Scope* visit(Scope* sc)
+    override void visit(CPPNamespaceDeclaration scd)
+    {   
+        auto scx = sc.copy();
+        scx.linkage = LINK.cpp;
+        scx.namespace = scd;
+        sc = scx;
+    }
+
+    override void visit(CPPMangleDeclaration cpmd)
+    {
+        sc = cpmd.createNewScope(sc, sc.stc, LINK.cpp, cpmd.cppmangle, sc.visibility, sc.explicitVisibility,
+            sc.aligndecl, sc.inlining);
+    }
+
+    override void visit(AlignDeclaration visd)
+    {
+     sc = visd.createNewScope(sc, sc.stc, sc.linkage, sc.cppmangle, sc.visibility, sc.explicitVisibility, visd, sc.inlining);
+    }
+
+    override void visit(PragmaDeclaration prd)
+    {
+        if (prd.ident == Id.Pinline)
+        {
+            // We keep track of this pragma inside scopes,
+            // then it's evaluated on demand in function semantic
+            sc = prd.createNewScope(sc, sc.stc, sc.linkage, sc.cppmangle, sc.visibility, sc.explicitVisibility, sc.aligndecl, prd); // @suppress(dscanner.style.long_line)
+        }
+    }
+
+override void visit(LinkDeclaration  lid)
+    {
+        sc= lid.createNewScope(sc, sc.stc, lid.linkage, sc.cppmangle, sc.visibility, sc.explicitVisibility,
+            sc.aligndecl, sc.inlining);
+    }
+
+/**
+     * Provides a new scope with `STC.deprecated_` and `Scope.depdecl` set
+     *
+     * Calls `StorageClassDeclaration.newScope` (as it must be called or copied
+     * in any function overriding `newScope`), then set the `Scope`'s depdecl.
+     *
+     * Returns:
+     *   Always a new scope, to use for this `DeprecatedDeclaration`'s members.
+     */
+    override void visit(DeprecatedDeclaration dpd)
+    {
+         auto oldsc = sc;
+        visit(cast(StorageClassDeclaration) dpd);
+        auto scx = sc;
+        sc = oldsc;
+        if (scx == sc) 
+        scx = sc.push(); 
+        scx.depdecl = dpd; 
+        sc = scx;
+    }
+    
+    /**************************************
+     * Use the ForwardingScopeDsymbol as the parent symbol for members.
+     */
+    override void visit(ForwardingAttribDeclaration  fad)
+    {
+        sc = sc.push(fad.sym);
+    }
+    
+ override void visit(StorageClassDeclaration swt)
+    {
+        StorageClass scstc = swt.stc;
+        /* These sets of storage classes are mutually exclusive,
+         * so choose the innermost or most recent one.
+         */
+        if (scstc & (STC.auto_ | STC.scope_ | STC.static_ | STC.extern_ | STC.manifest))
+            scstc &= ~(STC.auto_ | STC.scope_ | STC.static_ | STC.extern_ | STC.manifest);
```

Comment:
This is wrong. To make things clear, there are two `stc`s involved here:
 - StorageClassDeclaration.stc: this one is used simply as `stc` in the old code and should be replaced with `swt.stc` in new code.
 - sc.stc: this one should stay the same.

What you are doing is only getting `swt.stc` and using that everywhere.

The above if should be rewritten as:

```d
if (swt.stc & ....)
    scstc &= ...
```

Where `scstc = sc.stc` (as the name suggests)